data-prepper-plugins/date-processor/src/main/java/com/amazon/dataprepper/plugins/processor/date/DateProcessorConfig.java

```diff
@@ -42,11 +52,91 @@ public String getDestination() {
         return destination;
     }
 
-    public String getTimezone() {
-        return timezone;
+    public ZoneId getZonedId() {
+        return zoneId;
+    }
+
+    public Locale getSourceLocale() {
+        return sourceLocale;
+    }
+
+    private ZoneId buildZoneId(String timezone) {
+        try {
+            return ZoneId.of(timezone);
+        } catch (Exception e) {
+            throw new IllegalArgumentException("Invalid timezone.");
+        }
+    }
+
+    private Locale buildLocale(String locale) {
+        Locale currentLocale;
+        if (locale == null || locale.equalsIgnoreCase("ROOT")) {
+            return Locale.ROOT;
+        }
+
+        boolean isBCP47Format = locale.contains("-");
+
+        final String[] localeFields;
+        if (isBCP47Format) {
+            localeFields = locale.split("-");
+        }
+        else
+            localeFields = locale.split("_");
+
+        switch (localeFields.length) {
+            case 1:
+                currentLocale = new Locale(localeFields[0]);
+                break;
+            case 2:
+                currentLocale = new Locale(localeFields[0], localeFields[1]);
+                break;
+            case 3:
+                currentLocale = new Locale(localeFields[0], localeFields[1], localeFields[2]);
+                break;
+            default:
+                throw new IllegalArgumentException("Invalid locale format. Only language, country and variant are supported.");
+        }
+
+        if (currentLocale.getISO3Language() != null && currentLocale.getISO3Country() != null)
+            return currentLocale;
+        else
+            throw new IllegalArgumentException("Unknown locale provided.");
+    }
+
+    @AssertTrue(message = "match and from_time_received are mutually exclusive options.")
+    boolean isValidMatchAndFromTimestampReceived() {
+        return !Boolean.TRUE.equals(fromTimeReceived) || match == null;
```

Comment:
Yeah, this is something I discussed with @cmanning09 last week. Right now all the configurations options are optional. Should we change it to make sure customer configures either `fromTimeReceived` or `match`? If `fromTimeReceived` is false and `match` is null we don't do anything and just return the same event with current implementation.

Reply:
It could go either way, but I think I would rather enforce that the processor do something. A user may incorrectly expect that the processor should be working if we don't.

Reply:
@asifsmohammed, can you drive consensus on what the default behavior should be? I am in agreement with Taylor but we should confirm with the rest of the team.